homeassistant/components/alert/__init__.py

```diff
@@ -85,51 +90,68 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Alert component."""
     component = EntityComponent[Alert](LOGGER, DOMAIN, hass)
 
-    entities: list[Alert] = []
-
-    for object_id, cfg in config[DOMAIN].items():
-        if not cfg:
-            cfg = {}
-
-        name = cfg[CONF_NAME]
-        watched_entity_id = cfg[CONF_ENTITY_ID]
-        alert_state = cfg[CONF_STATE]
-        repeat = cfg[CONF_REPEAT]
-        skip_first = cfg[CONF_SKIP_FIRST]
-        message_template = cfg.get(CONF_ALERT_MESSAGE)
-        done_message_template = cfg.get(CONF_DONE_MESSAGE)
-        notifiers = cfg[CONF_NOTIFIERS]
-        can_ack = cfg[CONF_CAN_ACK]
-        title_template = cfg.get(CONF_TITLE)
-        data = cfg.get(CONF_DATA)
-
-        entities.append(
-            Alert(
-                hass,
-                object_id,
-                name,
-                watched_entity_id,
-                alert_state,
-                repeat,
-                skip_first,
-                message_template,
-                done_message_template,
-                notifiers,
-                can_ack,
-                title_template,
-                data,
+    @callback
+    def _async_load_alerts(config: ConfigType) -> list[Alert]:
+        """Load alert configuration."""
+        entities: list[Alert] = []
+        for object_id, cfg in config[DOMAIN].items():
+            if not cfg:
+                cfg = {}
```

Comment:
Code coverage was affected because of the refactoring, But it seems config can never be an empty `dict` because the schema has two required options `name` and `entity_id`. Any instance was validated against `ALERT_SCHEMA`.
This makes this code unreachable.

```suggestion
```